src/pop-state.js

```diff
@@ -0,0 +1,18 @@
+import { wrap } from 'lodash';
+
+let popState = false;
+
+if (typeof window !== 'undefined') {
+    window.addEventListener('popstate', () => {
+        popState = true;
+    });
+
+    history.pushState = wrap(history.pushState, (pushState, ...args) => {
+        popState = false;
+        pushState.apply(history, args);
+    });
+}
+
+const isPopState = () => popState;
```

Comment:
The function name seems odd. How about `isPopStateAvailable` ?

Reply:
I was struggling with the name of this one. `isPopStateAvailable` seems strange as there's nothing "available". 

The function returns true if the current history entry originated from a `popstate`, and false otherwise. Based on this, maybe `isHistoryEntryFromPopState()`?

Reply:
I like `isHistoryEntryFromPopState`.

Reply:
Will change.